skyportal/handlers/api/image_analysis.py

```diff
@@ -0,0 +1,422 @@
+            sim['flags_measured'][sidx] = obj1['flags'][oidx]
+
+            sims.append(sim)
+
+        sims = vstack(sims)
+
+        # FIXME
+        df = {
+            'ra': [ra0],
+            'dec': [dec0],
+            'magsys': ['ab'],
+            'mjd': [time.mjd],
+            'mag': [sims['mag_calib'][0]],
+            'magerr': [sims['magerr_measured'][0]],
+            'limiting_mag': [sims['mag_calib'][0]],
+            'filter': [filt],
+        }
+
+        data_out = {
+            'obj_id': obj_id,
+            'instrument_id': instrument.id,
+            'group_ids': [g.id for g in user.accessible_groups],
+            **df,
+        }
+
+        add_external_photometry(data_out, user)
+        os.remove(filename)
+    except Exception as e:
+        os.remove(filename)
+        raise e
+
+
+class ImageAnalysisHandler(BaseHandler):
+    @auth_or_token
+    def post(self, obj_id):
+        """
+        ---
+        description: Submit a new image for analysis
+        parameters:
+          - in: path
+            name: obj_id
+            required: true
+            schema:
+              type: string
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        try:
+            if cfg['image_analysis'] is not True:
+                return self.error('Image analysis is not enabled')
+
+            missing_bins = []
+            for exe in ['scamp', 'psfex']:
+                bin = shutil.which(exe)
+                if bin is None:
+                    missing_bins.append(exe)
+
+            if len(missing_bins) > 0:
+                return self.error(
+                    f"Can't run image analysis, missing dependencies: {', '.join(missing_bins)}"
+                )
+
+            data = self.get_json()
+            instrument_id = data.get('instrument_id')
+            if instrument_id is None:
+                return self.error(message=f'Missing instrument {instrument_id}')
+            instrument = Instrument.get_if_accessible_by(
+                instrument_id, self.current_user, raise_if_none=True, mode="read"
+            )
+
+            image_data = data.get("image_data")
+            if image_data is None:
+                return self.error(message='Missing image data')
+
+            filt = data.get("filter")
+            obstime = data.get("obstime")
+            obstime = Time(arrow.get(obstime.strip()).datetime)
+            image_data = image_data.split('base64,')
+            image_name = image_data[0].split('name=')[1].split(';')[0]
+            # if image name contains (.fz) then it is a compressed file
+            if image_name.endswith('.fz'):
+                suffix = '.fits.fz'
+            else:
+                suffix = '.fits'
+            with tempfile.NamedTemporaryFile(
```

Comment:
Should this be delete=True?

Reply:
So when I've set it to true, I ended up having issues down the line. Instead, I have another piece of code down the line that removes the file. But now that you mention it, I need to make sure that the file is deleted no matter what, which means adding that piece of code in the `expect` of the handler method. Let me take care of that before we merge anything.

Reply:
Either this or we move the whole creating and opening a file part inside of the reduce_image function (that runs in executor), and out of the handler.

Reply:
Here's what I'm scared of:
What happens when the app crashes/dies? Currently, even if I add some code to remove the file in the except statement, well naturally if the app is killed and restarted because it is not responsive, none of that code is reached. Therefore the file is not deleted...

Reply:
Maybe it's best that it put delete to True, and in the run in executor part, I recreate the file also with delete = True. The issue with putting delete=True only in the handler is, as it is asynchronous, when the handler returns it deletes the file and then the method that is running in executor can't open it!

Reply:
We can try it and see if we run into issues.

Reply:
I'll push some code super soon. Basically we were reopening later in some astropy methods to get the header. But we already have it! So I'm changing the reduce_image method to get rid of that.

Reply:
Turns out that sextractor and psfex also create temp directories and files, which are not deleted if the app dies, or even worse not delete at all if there is an exception...
So I'm adding some code to take care of that. I delete them when an exception occurs, and when the app restarts dies/restarts, it deletes all temp files related to sextractor or pfsex, ensuring that we don't accumulate useless data over time and fill up the disk space.

Reply:
Should be ok now